### PR TITLE
Fix `Proxy` is not set for `NodeRuntime`

### PR DESCRIPTION
**Pull request by JunkuiZhang**
Currently, on the `main` branch, the `proxy` in `NodeRuntime` is always set to `None`. This issue was unintentionally introduced in PR #15446.

Additionally, I noticed that under the changes in this PR, `HttpClientWithProxy` and `IsahcHttpClient` share the same member variables. Therefore, I have removed `HttpClientWithProxy` to avoid redundancy.

cc @osiewicz 

Release Notes:

- N/A

### CODE REVIEW

File: crates/isahc_http_client/src/isahc_http_client.rs [hunk 1]
```diff
@@ -54,6 +63,7 @@ impl HttpClient for IsahcHttpClient {
             let mut builder = isahc::Request::builder()
                 .method(parts.method)
                 .uri(parts.uri)
+                .proxy(self.proxy.clone())
```
Comment (JunkuiZhang) [inline crates/isahc_http_client/src/isahc_http_client.rs:66]: BTW, I dont know if this is needed, since we call `client_with_proxy.send_async(req)` later.

**Comment by maxdeviant** [comment]
> Therefore, I have removed `HttpClientWithProxy` to avoid redundancy.

Can we put this back? It's intended to help us abstract over HTTP clients.

**Comment by JunkuiZhang** [comment]
> Can we put this back? It's intended to help us abstract over HTTP clients.

With the changes in this PR, `HttpClientWithUrl` now includes the functionality of `HttpClientWithProxy`.

In this PR, the `proxy` information is stored in `dyn HttpClient`. If we were to keep `HttpClientWithProxy`, its structure would look like this:

```rust
struct HttpClientWithProxy(Arc<dyn HttpClient>);
```

Its methods would essentially be repetitive, just mirroring those of `dyn HttpClient`, which is why I think removing it makes the code less verbose.

However, if you prefer to keep it, I’m happy to modify the PR accordingly.

**Comment by JunkuiZhang** [comment]
Fixed via #19055

**Pull request closed without merging**